cli/cmd/common/chart_test.go

```diff
@@ -0,0 +1,56 @@
+package common
+
+import (
+	"embed"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+// Embed a test chart to test against.
+//go:embed fixtures/consul/* fixtures/consul/templates/_helpers.tpl
+var testChartFiles embed.FS
+
+func TestLoadChart(t *testing.T) {
+	directory := "fixtures/consul"
+
+	expectedApiVersion := "v2"
+	expectedName := "Foo"
+	expectedVersion := "0.1.0"
+	expectedDescription := "Mock Helm Chart for testing."
+	expectedValues := map[string]interface{}{
+		"key": "value",
+	}
+
+	actual, err := LoadChart(testChartFiles, directory)
+	require.NoError(t, err)
+	require.Equal(t, expectedApiVersion, actual.Metadata.APIVersion)
+	require.Equal(t, expectedName, actual.Metadata.Name)
+	require.Equal(t, expectedVersion, actual.Metadata.Version)
+	require.Equal(t, expectedDescription, actual.Metadata.Description)
+	require.Equal(t, expectedValues, actual.Values)
+}
+
+func TestReadChartFiles(t *testing.T) {
+	directory := "fixtures/consul"
+	expectedFiles := map[string]string{
+		"Chart.yaml":             "# This is a mock Helm Chart.yaml file used for testing.\napiVersion: v2\nname: Foo\nversion: 0.1.0\ndescription: Mock Helm Chart for testing.",
+		"values.yaml":            "# This is a mock Helm values.yaml file used for testing.\nkey: value",
+		"templates/_helpers.tpl": "helpers",
+		"templates/foo.yaml":     "foo: bar\n",
+	}
+
+	files, err := ReadChartFiles(testChartFiles, directory)
+	require.NoError(t, err)
+
+	actualFiles := make(map[string]string, len(files))
+	for _, f := range files {
+		actualFiles[f.Name] = string(f.Data)
+	}
+
+	for expectedName, expectedContents := range expectedFiles {
+		actualContents, ok := actualFiles[expectedName]
+		require.True(t, ok, "Expected file %s not found", expectedName)
+		require.Equal(t, expectedContents, actualContents)
+	}
+}
```

Comment:
Yes

Reply:
So I took some time to look into this test. It would require setting up a mock K8s client and loading it with configuration as part of the test setup. I think this would be a good test to do, but I feel we should first do some work to abstract over Helm in the code. Right now, the code is coupled to the Helm CLI code in a way that makes it hard to test. I have an idea for how to improve this. Are you ok with skipping this test for now?